### PR TITLE
Add CustomTagController

### PR DESCRIPTION
**Pull request by pengchen0692**
## Description
Add controller for custom tag

## Related issues
Addresses AB#78129

### CODE REVIEW

File: src/Microsoft.Health.Dicom.Core/Messages/CustomTag/AddCustomTagResponse.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        private readonly IEnumerable<CustomTagEntry> _customTags;
+        private readonly string _job;
+
+        public AddCustomTagResponse(IEnumerable<CustomTagEntry> customTags, string job)
+        {
+            _customTags = customTags;
+            _job = job;
+        }
+
+        public IEnumerable<CustomTagEntry> CustomTags { get => _customTags; }
```
Comment (brandonpollett) [inline src/Microsoft.Health.Dicom.Core/Messages/CustomTag/AddCustomTagResponse.cs:22]: nit: you can just convert these and the others like it to auto-properties

e.g. remove the private variables and replace this line with something like
```csharp
public IEnumerable<CustomTagEntry> CustomTags { get; }
```

Then in the constructor you can change `_customTags = customTags;` to `CustomTags = customTags`

File: src/Microsoft.Health.Dicom.Core/Messages/CustomTag/AddCustomTagResponse.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+namespace Microsoft.Health.Dicom.Core.Messages.CustomTag
+{
+    public class AddCustomTagResponse
+    {
+        private readonly IEnumerable<CustomTagEntry> _customTags;
+        private readonly string _job;
+
+        public AddCustomTagResponse(IEnumerable<CustomTagEntry> customTags, string job)
+        {
+            _customTags = customTags;
```
Comment (brandonpollett) [inline src/Microsoft.Health.Dicom.Core/Messages/CustomTag/AddCustomTagResponse.cs:18]: Should we ensure these aren't null/empty?

File: src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntry.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        /// <summary>
+        /// Key of this custom tag entry.
+        /// </summary>
+        public long Key { get; set; }
+
+        /// <summary>
+        /// Path of this tag. Normally it's composed of groupid and elementid.
+        /// E.g: 00100020 is path of patient id.
+        /// </summary>
+        public string Path { get; set; }
```
Comment (brandonpollett) [inline src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntry.cs:22]: Do we need to do any sort of validation on the `Path` or `VR` properties anywhere? I assume that that's there's some format we expect these in.

File: src/Microsoft.Health.Dicom.Core/Messages/CustomTag/AddCustomTagResponse.cs (hunk 2 above)
Comment (pengchen0692) [reply src/Microsoft.Health.Dicom.Core/Messages/CustomTag/AddCustomTagResponse.cs:18]: not sure about that. Checked several XXResponse (e.g: https://github.com/microsoft/dicom-server/blob/master/src/Microsoft.Health.Dicom.Core/Messages/Store/StoreResponse.cs), none of them validate the parameters.

Comment (pengchen0692) [reply src/Microsoft.Health.Dicom.Core/Messages/CustomTag/AddCustomTagResponse.cs:18]: My best guess is that these type of classes are data model, so any input should be acceptable. The validation could be in real business logic.

File: src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntry.cs (hunk 3 above)
Comment (pengchen0692) [reply src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntry.cs:22]: yes, CustomTagEntry is used when adding custom tag, the Path and VR will be validated over there.

File: src/Microsoft.Health.Dicom.Core/Messages/CustomTag/AddCustomTagResponse.cs (hunk 1 above)
Comment (pengchen0692) [reply src/Microsoft.Health.Dicom.Core/Messages/CustomTag/AddCustomTagResponse.cs:22]: resolved.

**Review by brandonpollett — APPROVED** [review]

**Pull request merged**